fcrypt/utils.go

```diff
@@ -18,15 +18,121 @@
 package fcrypt
 
 import (
+	"context"
+	"os"
 	"strings"
 
 	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/aoscloud/aos_common/api/cloudprotocol"
 )
 
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+// CryptoContext interface to access crypto functions.
+type CryptoContext interface {
+	ImportSessionKey(keyInfo CryptoSessionKeyInfo) (SymmetricContextInterface, error)
+	CreateSignContext() (SignContextInterface, error)
+}
+
+// CryptoContext contains necessary parameters for decryption.
+type DecryptParams struct {
+	CryptoContext  CryptoContext
```

Comment:
Can `CryptoContext`, `EncryptedFile` and `DecryptedFile` be moved as separate parameters? The following signature has better readability:
```go
func DecryptAndValidate(cryptoContext  CryptoContext, encryptedFile, decryptedFile string, params DecryptParams) error
```

Reply:
Done